hexl/eltwise/eltwise-reduce-mod-avx512.hpp

```diff
@@ -292,7 +296,73 @@ void EltwiseMontgomeryFormAVX512(uint64_t* result, const uint64_t* a,
     }
 
     __m512i v_c = _mm512_hexl_montgomery_reduce<BitShift, r>(
-        v_T_hi, v_T_lo, v_modulus, v_inv_mod, v_prod_rs);
+        v_T_hi, v_T_lo, v_modulus, v_neg_inv_mod, v_prod_rs);
+    HEXL_CHECK_BOUNDS(ExtractValues(v_c).data(), 8, modulus,
+                      "v_op exceeds bound " << modulus);
+    _mm512_storeu_si512(v_result, v_c);
+    ++v_a;
+    ++v_result;
+  }
+}
+
+/// @brief Convert out of the Montgomery Form computed via the REDC algorithm,
+/// also known as Montgomery reduction.
+/// @tparam BitShift denotes the operational length, in bits, of the operands
+/// and result values.
+/// @tparam r defines the value of R, being R = 2^r. R > modulus.
+/// @param[in] a input vector in Montgomery Form.
+/// @param[in] modulus such that gcd(R, modulus) = 1.
+/// @param[in] neg_inv_mod in [0, R − 1] such that q*neg_inv_mod ≡ −1 mod R,
+/// @param[in] n number of elements in input vector.
+/// @param[out] result unsigned long int vector in the range [0, q − 1] such
+/// that S ≡ TR^−1 mod q
+template <int BitShift, int r>
+void EltwiseMontgomeryFormOutAVX512(uint64_t* result, const uint64_t* a,
+                                    uint64_t n, uint64_t modulus,
+                                    uint64_t neg_inv_mod) {
+  HEXL_CHECK(a != nullptr, "Require operand a != nullptr");
+  HEXL_CHECK(n != 0, "Require n != 0");
+  HEXL_CHECK(modulus > 1, "Require modulus > 1");
+
+  uint64_t R = (1ULL << r);
+  HEXL_CHECK(std::__gcd(static_cast<int64_t>(modulus), static_cast<int64_t>(R)),
```

Comment:
`__gcd` is perhaps okay for now, but doesn't sound ideal (see https://stackoverflow.com/a/68772237). Perhaps time to consider upgrading to C++17 so we can use `std::gcd`